### PR TITLE
check http response codes in more places

### PR DESCRIPTION
**Pull request by minrk**
check for error codes with `response.raise_for_status()` before assuming responses are valid JSON, etc.

related to #498 where there's a pretty good chance that travis is responding with an error.

### CODE REVIEW

File: conda_smithy/ci_register.py [hunk 1]
```diff
@@ -188,8 +188,9 @@ def add_project_to_travis(user, project):
 
         response = requests.get(url, headers=headers)
         try:
+            response.raise_for_status()
             content = response.json()
-        except ValueError:
+        except Exception:
```
Comment (jakirkham) [inline conda_smithy/ci_register.py]: Couldn't we do better than a generic exception here? Looks like this should be an [`HTTPError`]( http://docs.python-requests.org/en/master/api/#requests.Response.raise_for_status ) for the status. I don't know if the `ValueError` from trying to parse the JSON still needs to be handled, but I guess we could do that too if needed.

**Comment by jakirkham** [comment]
Thanks @minrk. Barring one minor comment, LGTM.

File: conda_smithy/ci_register.py (hunk 1 above)
Comment (minrk) [inline conda_smithy/ci_register.py]: Good call, done.

**Comment by jakirkham** [comment]
Mind taking a quick look, @ericdill?

File: conda_smithy/ci_register.py [hunk 2]
```diff
@@ -188,8 +188,9 @@ def add_project_to_travis(user, project):
 
         response = requests.get(url, headers=headers)
         try:
+            response.raise_for_status()
             content = response.json()
-        except ValueError:
+        except (requests.HTTPError, ValueError):
```
Comment (ericdill) [inline conda_smithy/ci_register.py]: Ah, yes of course. That's a critical addition here

**Comment by jakirkham** [comment]
Is there a way for us to tell Travis CI to send the response in chunks? Maybe trying to send a big glob of data is causing the failure.

**Comment by minrk** [comment]
> Is there a way for us to tell Travis CI to send the response in chunks?

I don't believe so. GitHub's API has a paged-result structure for all list results, but the Travis API only appears to have a similar limit for logs and requests (basically the endpoints they expect to have a lot of results). Looks like #505 circumvents the problem, anyway.

**Pull request merged**